rskj-core/src/test/java/co/rsk/peg/lockingcap/LockingCapStorageProviderImplTest.java

```diff
@@ -0,0 +1,114 @@
+        Coin previousLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(previousLockingCap);
+        lockingCapStorageProvider.save(activations);
+        // Recreate LockingCapStorageProvider to load the previous value from storage
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+
+        // Assert
+        assertEquals(Optional.of(previousLockingCap), actualLockingCap);
+    }
+
+    @Test
+    void getLockingCap_prePapyrus200_whenLockingCapIsNotSet_shouldReturnEmpty() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+
+        // Act
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+
+        // Assert
+        assertEquals(Optional.empty(), actualLockingCap);
+    }
+
+    @Test
+    void setLockingCap_whenIsSetANewLockingCapValue_shouldSetLockingCap() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+
+        // Act
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Assert
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+        assertEquals(Optional.of(newLockingCap), actualLockingCap);
+    }
+
+    @Test
+    void save_whenIsSavedANewLockingCapValue_shouldSaveLockingCap() {
+        // Arrange
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Act
+        lockingCapStorageProvider.save(activations);
+
+        // Assert
+        // Recreate LockingCapStorageProvider to load the previous value from storage and make sure it was saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+        assertEquals(Optional.of(newLockingCap), actualLockingCap);
+    }
+
+    @Test
+    void save_prePapyrus200_whenIsAttemptedSaveANewLockingCapValue_shouldNotSaveLockingCap() {
+        // Arrange
+        activations = ActivationConfigsForTest.wasabi100().forBlock(0);
+        Coin newLockingCap = constants.getInitialValue().add(Coin.SATOSHI);
+        lockingCapStorageProvider.setLockingCap(newLockingCap);
+
+        // Act
+        lockingCapStorageProvider.save(activations);
+
+        // Assert
+        // Recreate LockingCapStorageProvider to load the previous value from storage and make sure it was not saved
+        lockingCapStorageProvider = new LockingCapStorageProviderImpl(bridgeStorageAccessor);
+        Optional<Coin> actualLockingCap = lockingCapStorageProvider.getLockingCap(activations);
+        assertEquals(Optional.empty(), actualLockingCap);
```

Comment:
same as above, check the value got instead of the Optional

Reply:
Now it looks like this: `assertEquals(expectedLockingCap, actualLockingCap);`

We can use Optional values in tests without using the get() method and avoid adding additional lines to ensure isPresent() the value. The test has an expected result and should assert correctly what we are testing.